docs/PID.md

```diff
@@ -0,0 +1,109 @@
+# PID
+PID control is a widely used control methodology in the 3d printing world.
+It’s ubiquitous when it comes to temperature control, be it with heaters for
+generating heat, or fans for removing it. The goal of this document is to
+provide a high level overview of what PID is, and how to best leverage it in
+Klipper.
+
+### History
+
+The first rudimentary PID controller was developed by Elmer Sperry in 1911 to
+automate a ship's rudder control. Engineer Nicolas Minorsky published the first
+mathematical analysis of a PID controller in 1922. In 1942 John Ziegler &
+Nathaniel Nichols published their groundbreaking paper
+“Optimum Settings for Automatic Controllers” that layed out a trial and error
+method for tuning a pid controller commonly referred to as the
+“Ziegler-Nichols Method” today.
+
+In 1984 Karl Astrom and Tore Hagglund published their paper “Automatic Tuning
+of Simple Regulators with Specifications on Phase and Amplitude Margins”. In
+the paper they introduced an automated tuning methodology commonly referred to
+as the “Astrom-Hagglund method” or the “Relay method”.
+
+In 2019 Brandon Taysom & Carl Sorensen published their paper "Adaptive Relay
+Autotuning under Static and Non-static Disturbances with Application to
+Friction Stir Welding", that laid out a method to generate more accurate
+results from a relay test. This is the PID calibration method currently used by
+Klipper.
+
+### PID Calibration
+
+As previously mentioned Klipper uses a relay test for calibration purposes. A
+standard relay test is conceptually simple. You turn the heater's power on and
+off to get it to oscillate about the target temperature, as seen in the
+following plot.
+
+![simple relay test](img/pid_01.png)
+
+The above plot shows a common issue with a standard relay test. If the system
+being calibrated has too much or too little power for the chosen target
+temperature, it will produce biased and asymmetric results. As the plot above
+shows, the system spends more time with the power off, and has a greater
+amplitude above the target temperature than it does below.
```

Comment:
```suggestion
The above graph shows a common issue with a standard relay test. If the system
being calibrated has too much or too little power for the chosen target
temperature, it will produce biased and asymmetric results. As can be seen
above, the system spends more time in the off state than on and has a larger
amplitude above the target temperature than below.
```